src/arguments/CoreDate.ts

```diff
@@ -1,32 +1,27 @@
 import type { PieceContext } from '@sapphire/pieces';
 import { Identifiers } from '../lib/errors/Identifiers';
+import { resolveDate } from '../lib/resolvers';
 import { Argument, ArgumentContext, ArgumentResult } from '../lib/structures/Argument';
 
 export class CoreArgument extends Argument<Date> {
+	private readonly messages = {
+		[Identifiers.ArgumentDateTooSmall]: 'The argument is too small.',
+		[Identifiers.ArgumentDateTooBig]: 'The argument is too big.',
+		[Identifiers.ArgumentDateError]: 'The argument did not resolve to a date.'
```

Comment:
```suggestion
		[Identifiers.ArgumentDateTooEarly]: 'The given date is below the minimum allowed date.',
		[Identifiers.ArgumentDateTooFar]: 'The given date is beyond the maximum allowed date.',
		[Identifiers.ArgumentDateError]: 'The argument did not resolve to a valid date.'
```

And update `resolveDate` accordingly

Reply:
Dates do not have a notion of being too far, or below/beyond either. It should either be too early or too late.